### PR TITLE
refactor(api): define status phase type

### PR DESCRIPTION
**Pull request by supereagle**
**What this PR does / why we need it**:

Add your description

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @cd1989  @zhujian7 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/apis/cyclone/v1alpha1/workflow_run.go [hunk 1]
```diff
@@ -109,7 +112,7 @@ type PodInfo struct {
 // +k8s:deepcopy-gen=true
 type Status struct {
 	// Status with value: Running, Waiting, Completed, Error
-	Status string `json:"status"`
+	Status StatusPhase `json:"status"`
```
Comment (supereagle) [inline pkg/apis/cyclone/v1alpha1/workflow_run.go]: Change to `Phase`?

Comment (zhujian7) [reply pkg/apis/cyclone/v1alpha1/workflow_run.go]: Why need this change?

Comment (supereagle) [reply pkg/apis/cyclone/v1alpha1/workflow_run.go]: `Status.Status` is a little confused.

**Comment by cd1989** [comment]
It should not be `feat`

File: pkg/apis/cyclone/v1alpha1/workflow_run.go (hunk 1 above)
Comment (zhujian7) [reply pkg/apis/cyclone/v1alpha1/workflow_run.go]: ok

**Comment by zhujian7** [comment]
> It should not be `feat`

`refactor` is more precise?

**Comment by supereagle** [comment]
> `refactor` is more precise?

Done

**Pull request merged**